subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/DefaultInstantExecution.kt

```diff
@@ -487,18 +486,48 @@ class DefaultInstantExecution internal constructor(
     private
     val instantExecutionStateFile by unsafeLazy {
         val cacheDir = absoluteFile(".instant-execution-state/${currentGradleVersion()}")
-        val baseName = compactMD5For(startParameter.requestedTaskNames)
+        val baseName = createCompactMD5(instantExecutionCacheKey())
         val cacheFileName = "$baseName.bin"
         File(cacheDir, cacheFileName)
     }
 
+    private
+    fun instantExecutionCacheKey() = startParameter.run {
+        val cacheKey = StringBuilder()
+        requestedTaskNames.joinTo(cacheKey, separator = "/")
+        if (excludedTaskNames.isNotEmpty()) {
+            excludedTaskNames.joinTo(cacheKey, prefix = "-", separator = "/")
```

Comment:
`-` is a valid task name char, maybe use `:` or any other invalid chars instead to separate requested task names and excluded task names.

FTR
```
The task name 'foo:' must not contain any of the following characters: [/, \, :, <, >, ", ?, *, |].
```

Reply:
Yeah, better. Thanks!